packages/client/src/forms/utils.ts

```diff
@@ -543,25 +550,53 @@ export function getQueryData(
 
 export const getConditionalActionsForField = (
   field: IFormField,
-  /*
-   * These are used in the eval expression
-   */
   values: IFormSectionData,
-  offlineCountryConfig?: IOfflineData,
-  draftData?: IFormData,
-  userDetails?: UserDetails | null
+  offlineCountryConfig: IOfflineData,
+  draftData: IFormData,
+  userDetails: UserDetails | null
 ): string[] => {
   if (!field.conditionals) {
     return []
   }
   return (
     field.conditionals
       // eslint-disable-next-line no-eval
-      .filter((conditional) => eval(conditional.expression))
+      .filter((conditional) =>
+        evalExpressionInFieldDefinition(
+          conditional.expression,
+          values,
+          offlineCountryConfig,
+          draftData,
+          userDetails
+        )
+      )
       .map((conditional: Conditional) => conditional.action)
   )
 }
 
+export const evalExpressionInFieldDefinition = (
+  expression: string,
+  /*
+   * These are used in the eval expression
+   */
+  $form: IFormSectionData,
+  $config: IOfflineData,
+  $draft: IFormData,
+  $user: (UserDetails & { token?: string }) | null
+) => {
+  $user = $user !== null ? { ...$user, token: getToken() } : $user
```

Comment:
Can we make the `Authorization: <userToken>` automatic so the form configuring user doesn't have to even care about that? So request is always made as the currently logged in user?

Reply:
I guess then I should keep the auth header always with the request, isn't it @rikukissa ?